test/alt.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
```

Comment:
Pretty much. I was thinking we could use the free structures (free monoids, applicatives, monads ...). You do need a base functor for those, though, and I'm not sure whether the natural choice (`Identity`) actually gives you much assurance.